shared/src/question-generators/math/modTricks.ts

```diff
@@ -0,0 +1,292 @@
+import { Language } from "@shared/api/Language";
+import {
+    FreeTextFeedbackFunction,
+    FreeTextQuestion,
+    QuestionGenerator,
+} from "@shared/api/QuestionGenerator";
+import { serializeGeneratorCall } from "@shared/api/QuestionRouter";
+import Random from "@shared/utils/random";
+import { t, tFunctional, Translations } from "@shared/utils/translations";
+import {
+    gcd,
+    calculateModularInverse,
+    modularExponentiation,
+    solveCRT,
+    areCoprime
+} from "@shared/question-generators/math/utils";
+
+const translations: Translations = {
+    en: {
+        name: "Modular Arithmetic Tricks",
+        description: "Answer questions involving modular arithmetic.",
+        reductionQuestion: "Reduce ${{x}}$ modulo ${{n}}$.",
+        inverseQuestion: "Find the modular inverse of ${{a}}$ modulo ${{n}}$.",
+        expQuestion: "Calculate ${{a}}^{{{b}}} \\pmod{ {{n}} }$.",
+        simpleQuestion: "Find an integer $x$ such that $x ≡ {{a}} \\pmod{ {{b}} }$.",
+        crtQuestion: "Solve the system of congruences: \\[ {{text}} \\text{.}\\]",
+        crtBottomText: "Provide your answer in the form: $y\\pmod{z}$.",
+        feedbackInvalid: "Your answer is not valid.",
+        feedbackCorrect: "Correct!",
+        feedbackIncorrect: "Incorrect.",
+    },
+    de: {
+        name: "Modulare Arithmetik Tricks",
+        description: "Beantworten Sie Fragen zur modularen Arithmetik.",
+        reductionQuestion: "Reduzieren Sie ${{x}}$ modulo ${{n}}$.",
+        inverseQuestion: "Finden Sie das Inverse von ${{a}}$ modulo ${{n}}$.",
+        expQuestion: "Berechnen Sie ${{a}}^{{{b}}} \\pmod{ {{n}} }$.",
+        simpleQuestion: "Finden Sie eine ganze Zahl $x$, so dass $x ≡ {{a}} \\pmod{ {{b}} }$.",
+        crtQuestion: "Lösen Sie das System von Kongruenzen: \\[ {{text}} \\text{.}\\]",
+        crtBottomText: "Geben Sie Ihre Antwort in der Form $y\\pmod{z}$ an.",
+        feedbackInvalid: "Ihre Antwort ist ungültig.",
+        feedbackCorrect: "Richtig!",
+        feedbackIncorrect: "Falsch.",
+    }
+};
+
+export const ModTricks: QuestionGenerator = {
+    id: "modtricks",
+    name: tFunctional(translations, "name"),
+    description: tFunctional(translations, "description"),
+    tags: ["basic math", "modular arithmetic", "modulus", "mod", "arithmetic"],
+    languages: ["en", "de"],
+    author: "Janette Welker",
+    license: "MIT",
+    expectedParameters: [
+        {
+            type: "string",
+            name: "variant",
+            allowedValues: ["simple", "reduction", "inverse", "exponentiation", "crt"],
```

Comment:
The variant `crt` is a little more complex / different to the other ones. Maybe we can separate this into an own generator with variants for different difficulties, because congruent system of 4 (for me) is quite difficult. Just an idea

Reply:
Yes, would make sense.